### PR TITLE
use libgiza threading for staging code

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

**Comment by tychoish** [comment]
@i80and for your consideration. This change is hidden behind a flag for safety and testing (though there is some risk to the current implementation.) If this works and tests out, I think there's a fair amount of fun deletion possible :) 

Not required, but a thought.

File: giza/giza/operations/stage.py [hunk 1]
```diff
@@ -350,34 +381,40 @@ def stage(self, root):
...
+                logger.info('Uploading %s', local_path)
+                k.set_contents_from_filename(src_path, **Staging.S3_OPTIONS)
+
+                # And then copy it to its final destination
+                k.copy(bucket.name, full_name, **Staging.S3_OPTIONS)
+            else:
+                raise err
+    except boto.exception.S3ResponseError as err:
+        if should_retry is True:
+            logger.error("encountered error uploading {0}, retrying once in a bit.".format())
```
Comment (i80and) [inline giza/giza/operations/stage.py]: This should probably be log level "info" (or even "debug"), not "error". Getting this a couple times per stage is perfectly normal.

File: giza/giza/operations/stage.py [hunk 2]
```diff
@@ -350,34 +381,40 @@ def stage(self, root):
...
+            time.sleep(.25)
```
Comment (i80and) [inline giza/giza/operations/stage.py]: Stylistically (and this is SUPER nit-picky), I prefer "0.25" to ".25" since it's harder to misread.

**Comment by tychoish** [comment]
updated as discussed.

**Comment by tychoish** [comment]
I still like this, but I think after two years, it's not worth it!

**Comment by i80and** [comment]
Haha, sorry Sam! I never quiiiite got around to integrating it.

**Pull request closed without merging**